src/sage/rings/species.py

```diff
@@ -0,0 +1,2255 @@
+
+            sage: a = C(PermutationGroup([[(1,3),(2,4)], [(1,4),(2,3)]]))
+            sage: b = C(PermutationGroup([[(1,3,2,4)]]))
+            sage: a == b
+            False
+        """
+        def is_equal(elm):
+            return SymmetricGroup(elm._C.degree()).are_conjugate(elm._C, C)
+
+        def standardize(C):
+            if not C.degree():
+                return SymmetricGroup(0)
+            # an directly indecomposable group is transitive, so we
+            # can use the gap group without worrying about the domain
+            G = C.gap()
+            sorted_orbits = sorted(G.Orbits().sage(), key=len, reverse=True)
+            pi = PermutationGroupElement([e for o in sorted_orbits for e in o])
+            G = PermutationGroup(G.SmallGeneratingSet().sage())
+            return G.conjugate(pi.inverse())
+
+        key = cls._invariant(C)
+        if key in parent._cache:
+            lookup = parent._cache[key]
+            for elm in lookup:
+                if is_equal(elm):
+                    return elm
+        else:
+            lookup = parent._cache[key] = []
+
+        elm = WithPicklingByInitArgs.__classcall__(cls, parent, standardize(C))
+        lookup.append(elm)
+        return elm
+
+    def __hash__(self):
+        """
+        Return a hash of ``self``.
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import ConjugacyClassesOfDirectlyIndecomposableSubgroups
+            sage: C = ConjugacyClassesOfDirectlyIndecomposableSubgroups()
+            sage: G = PermutationGroup([[(1,2),(3,4)]])
+            sage: H = PermutationGroup([[(1,3),(2,4)], [(1,2),(3,4)]])
+            sage: K = PermutationGroup([[(1,2,3,4)]])
+            sage: hash(C(G)) == hash(C(H))
+            False
+            sage: hash(C(H)) == hash(C(K))
+            True
+            sage: C(H) == C(K)
+            False
+        """
+        return hash(self._invariant(self._C))
+
+    def __init__(self, parent, C):
+        r"""
+        Initialize a conjugacy class of directly indecomposable subgroups
+        of a symmetric group.
+
+        TESTS::
+
+            sage: from sage.rings.species import ConjugacyClassesOfDirectlyIndecomposableSubgroups
+            sage: C = ConjugacyClassesOfDirectlyIndecomposableSubgroups()
+            sage: G = C(PermutationGroup([[(1,2),(3,4)],[(1,2),(5,6)]]))
+            sage: TestSuite(G).run()
+        """
+        Element.__init__(self, parent)
+        self._C = C
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import ConjugacyClassesOfDirectlyIndecomposableSubgroups
+            sage: C = ConjugacyClassesOfDirectlyIndecomposableSubgroups()
+            sage: G = PermutationGroup([[(1,3),(4,7)], [(2,5),(6,8)], [(1,4),(2,5),(3,7)]])
+            sage: C(G)
+            ((5,6)(7,8), (1,2)(3,4), (1,3)(2,4)(5,6))
+        """
+        return f"{self._C.gens()}"
```

Comment:
```suggestion
        return repr(self._C.gens())
```
I find this easier to understand.